x/gamm/pool-models/stableswap/amm.go

```diff
@@ -288,8 +288,10 @@ func (p *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sdk
 		paCopy.updatePoolForJoin(sdk.NewCoins(tokenIn), newShares)
 		return &paCopy
 	}
-	// TODO: Correctly handle swap fee
-	return cfmm_common.BinarySearchSingleAssetJoin(p, tokenIn, poolWithAddedLiquidityAndShares)
+
+	tokenInAmtAfterFee := tokenIn.Amount.ToDec().Mul(sdk.OneDec().Sub(swapFee)).TruncateInt()
```

Comment:
nit: can we add a comment about the operations/math here?

Reply:
Added!